packages/react-next/src/components/Coachmark/Coachmark.base.tsx

```diff
@@ -270,58 +246,50 @@ function useProximityHandlers(
   translateAnimationContainer: React.RefObject<HTMLDivElement>,
   openCoachmark: () => void,
 ) {
-  const async = useAsync();
+  const { setTimeout, clearTimeout } = useSetTimeout();
 
-  /**
-   * The target element the mouse would be in
-   * proximity to
-   */
+  /** The target element the mouse would be in proximity to */
   const targetElementRect = React.useRef<DOMRect>();
 
-  const setTargetElementRect = (): void => {
-    if (translateAnimationContainer?.current) {
-      targetElementRect.current = translateAnimationContainer.current.getBoundingClientRect();
-    }
-  };
-
   React.useEffect(() => {
+    const setTargetElementRect = (): void => {
+      if (translateAnimationContainer.current) {
+        targetElementRect.current = translateAnimationContainer.current.getBoundingClientRect();
+      }
+    };
+
     const events = new EventGroup({});
 
-    // We don't want to the user to immediately trigger the Coachmark when it's opened
-    async.setTimeout(() => {
+    // We don't want the user to immediately trigger the Coachmark when it's opened
+    setTimeout(() => {
       const { mouseProximityOffset = 0 } = props;
-      /**
-       * An array of cached ids returned when setTimeout runs during
-       * the window resize event trigger.
-       */
+
+      /** Cached ids returned when setTimeout runs during the window resize event trigger. */
       const timeoutIds: number[] = [];
 
-      // Take the initial measure out of the initial render to prevent
-      // an unnecessary render.
-      async.setTimeout(() => {
+      // Take the initial measure out of the initial render to prevent an unnecessary render.
+      setTimeout(() => {
         setTargetElementRect();
 
-        // When the window resizes we want to async
-        // get the bounding client rectangle.
-        // Every time the event is triggered we want to
-        // setTimeout and then clear any previous instances
-        // of setTimeout.
+        // When the window resizes we want to async get the bounding client rectangle.
+        // Every time the event is triggered we want to setTimeout and then clear any previous
+        // instances of setTimeout.
         events.on(window, 'resize', (): void => {
           timeoutIds.forEach((value: number): void => {
-            clearInterval(value);
+            clearTimeout(value);
           });
+          timeoutIds.splice(0, timeoutIds.length); // clear array
```

Comment:
Without this, the array of timeouts keeps unnecessarily growing (and each one is unnecessarily cleared again) with every resize event